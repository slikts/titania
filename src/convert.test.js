const {
  convert,
  expandRow,
  splitRow,
  fillExpandedRow,
  convertLastCells,
} = require("./convert");

describe("convert", () => {
  it("splits row", () => {
    expect(splitRow(["a", "b", "\n"])).toEqual([["a"], ["b"], [""]]);
  });

  it("expands row", () => {
    expect(expandRow(["a", "b\nc", "\n\nd"])).toEqual([
      ["a", "b", ""],
      ["", "c", "d"],
    ]);
    expect(expandRow(["a", "b\nc\ne", "\n\nd\nf"])).toEqual([
      ["a", "b", ""],
      ["", "c", "d"],
      ["", "e", "f"],
    ]);

    expect(expandRow(["a", "b", "\n"])).toEqual([["a", "b", ""]]);
  });

  it("fills expanded row", () => {
    expect(
      fillExpandedRow([
        ["a", "b", ""],
        ["", "c", "d"],
      ])
    ).toEqual([
      ["a", "b", ""],
      ["a", "c", "d"],
    ]);

    expect(
      fillExpandedRow([
        ["a", "b", ""],
        ["", "", "d"],
      ])
    ).toEqual([
      ["a", "b", ""],
      ["a", "b", "d"],
    ]);
  });

  it("converts last cells", () => {
    expect(convertLastCells(["a", "b", "1.2.3", "4.5.6"])).toEqual([
      "a",
      "b",
      "=DATE(3, 2, 1)",
      "=DATE(6, 5, 4)",
    ]);
  });

  it("converts", () => {
    expect(convert(fixture)).toMatchSnapshot();
  });
});

const fixture = {
  "Ārlietu komisija": [
    ["Vārds, uzvārds", "Nosaukums", "Amats", "No", "Līdz"],
    [
      "Rihards Kols",
      "NA",
      "Priekšsēdētājs\nDeputāts",
      "23.11.2022\n23.11.2022",
      "\n\n23.11.2022",
    ],
    [
      "Irma Kalniņa",
      "JV",
      "Priekšsēdētāja biedre\nSekretāre\nDeputāte",
      "27.09.2023\n23.11.2022\n23.11.2022",
      "\n\n27.09.2023\n23.11.2022",
    ],
    [
      "Juris Viļums",
      "AS",
      "Sekretārs\nPriekšsēdētāja biedrs\nDeputāts",
      "27.09.2023\n23.11.2022\n23.11.2022",
      "\n\n27.09.2023\n23.11.2022",
    ],
    ["Edmunds Cepurītis", "PRO", "Deputāts", "21.09.2023", "\n"],
    ["Jānis Dombrava", "NA", "Deputāts", "21.09.2023", "\n"],
    ["Zanda Kalniņa-Lukaševica", "JV", "Deputāte", "23.11.2022", "\n"],
    ["Dmitrijs Kovaļenko", "ST!", "Deputāts", "23.11.2022", "\n"],
    ["Edvards Smiltēns", "AS", "Deputāts", "23.11.2022", "\n"],
    ["Ričards Šlesers", "LPV", "Deputāts", "23.11.2022", "\n"],
    ["Jānis Vucāns", "ZZS", "Deputāts", "21.09.2023", "\n"],
  ],
  "Budžeta un finanšu (nodokļu) komisija": [
    ["Vārds, uzvārds", "Nosaukums", "Amats", "No", "Līdz"],
    [
      "Jānis Reirs",
      "JV",
      "Priekšsēdētājs\nDeputāts",
      "15.12.2022\n15.12.2022",
      "\n\n15.12.2022",
    ],
    [
      "Andris Šuvajevs",
      "PRO",
      "Priekšsēdētāja biedrs\nDeputāts",
      "27.09.2023\n23.11.2022",
      "\n\n27.09.2023",
    ],
    [
      "Aiva Vīksna",
      "AS",
      "Sekretāre\nDeputāte",
      "23.11.2022\n23.11.2022",
      "\n\n23.11.2022",
    ],
    [
      "Artūrs Butāns",
      "NA",
      "Deputāts\nPriekšsēdētāja biedrs\nDeputāts",
      "27.09.2023\n23.11.2022\n23.11.2022",
      "\n\n27.09.2023\n23.11.2022",
    ],
    ["Raimonds Čudars", "JV", "Deputāts", "21.09.2023", "\n"],
  ],
};
