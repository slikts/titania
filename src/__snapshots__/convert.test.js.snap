// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert converts 1`] = `
[
  [
    "Komisija",
    "Vārds, uzvārds",
    "Partija",
    "Amats",
    "No",
    "Līdz",
  ],
  [
    "Ārlietu komisija",
    "Rihards Kols",
    "NA",
    "Priekšsēdētājs",
    "=DATE(2022, 11, 23)",
    "",
  ],
  [
    "Ārlietu komisija",
    "Rihards Kols",
    "NA",
    "Deputāts",
    "=DATE(2022, 11, 23)",
    "=DATE(2022, 11, 23)",
  ],
  [
    "Ārlietu komisija",
    "Irma Kalniņa",
    "JV",
    "Priekšsēdētāja biedre",
    "=DATE(2023, 09, 27)",
    "",
  ],
  [
    "Ārlietu komisija",
    "Irma Kalniņa",
    "JV",
    "Sekretāre",
    "=DATE(2022, 11, 23)",
    "=DATE(2023, 09, 27)",
  ],
  [
    "Ārlietu komisija",
    "Irma Kalniņa",
    "JV",
    "Deputāte",
    "=DATE(2022, 11, 23)",
    "=DATE(2022, 11, 23)",
  ],
  [
    "Ārlietu komisija",
    "Juris Viļums",
    "AS",
    "Sekretārs",
    "=DATE(2023, 09, 27)",
    "",
  ],
  [
    "Ārlietu komisija",
    "Juris Viļums",
    "AS",
    "Priekšsēdētāja biedrs",
    "=DATE(2022, 11, 23)",
    "=DATE(2023, 09, 27)",
  ],
  [
    "Ārlietu komisija",
    "Juris Viļums",
    "AS",
    "Deputāts",
    "=DATE(2022, 11, 23)",
    "=DATE(2022, 11, 23)",
  ],
  [
    "Ārlietu komisija",
    "Edmunds Cepurītis",
    "PRO",
    "Deputāts",
    "=DATE(2023, 09, 21)",
    "",
  ],
  [
    "Ārlietu komisija",
    "Jānis Dombrava",
    "NA",
    "Deputāts",
    "=DATE(2023, 09, 21)",
    "",
  ],
  [
    "Ārlietu komisija",
    "Zanda Kalniņa-Lukaševica",
    "JV",
    "Deputāte",
    "=DATE(2022, 11, 23)",
    "",
  ],
  [
    "Ārlietu komisija",
    "Dmitrijs Kovaļenko",
    "ST!",
    "Deputāts",
    "=DATE(2022, 11, 23)",
    "",
  ],
  [
    "Ārlietu komisija",
    "Edvards Smiltēns",
    "AS",
    "Deputāts",
    "=DATE(2022, 11, 23)",
    "",
  ],
  [
    "Ārlietu komisija",
    "Ričards Šlesers",
    "LPV",
    "Deputāts",
    "=DATE(2022, 11, 23)",
    "",
  ],
  [
    "Ārlietu komisija",
    "Jānis Vucāns",
    "ZZS",
    "Deputāts",
    "=DATE(2023, 09, 21)",
    "",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Jānis Reirs",
    "JV",
    "Priekšsēdētājs",
    "=DATE(2022, 12, 15)",
    "",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Jānis Reirs",
    "JV",
    "Deputāts",
    "=DATE(2022, 12, 15)",
    "=DATE(2022, 12, 15)",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Andris Šuvajevs",
    "PRO",
    "Priekšsēdētāja biedrs",
    "=DATE(2023, 09, 27)",
    "",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Andris Šuvajevs",
    "PRO",
    "Deputāts",
    "=DATE(2022, 11, 23)",
    "=DATE(2023, 09, 27)",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Aiva Vīksna",
    "AS",
    "Sekretāre",
    "=DATE(2022, 11, 23)",
    "",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Aiva Vīksna",
    "AS",
    "Deputāte",
    "=DATE(2022, 11, 23)",
    "=DATE(2022, 11, 23)",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Artūrs Butāns",
    "NA",
    "Deputāts",
    "=DATE(2023, 09, 27)",
    "",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Artūrs Butāns",
    "NA",
    "Priekšsēdētāja biedrs",
    "=DATE(2022, 11, 23)",
    "=DATE(2023, 09, 27)",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Artūrs Butāns",
    "NA",
    "Deputāts",
    "=DATE(2022, 11, 23)",
    "=DATE(2022, 11, 23)",
  ],
  [
    "Budžeta un finanšu (nodokļu) komisija",
    "Raimonds Čudars",
    "JV",
    "Deputāts",
    "=DATE(2023, 09, 21)",
    "",
  ],
]
`;
